crm_event/models/event_type.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+
+
+class EventType(models.Model):
+    _inherit = "event.type"
+
+    seats_available_total = fields.Char(
+        string="Events available (and seats)",
+        compute="_compute_event_totals",
+        help="Upcoming/running events of this category (and available seats).",
+    )
+    crm_lead_ids = fields.One2many(
+        string="Leads/Opportunities",
+        comodel_name="crm.lead",
+        inverse_name="event_type_id",
+    )
+    open_opportunities_count = fields.Integer(
+        compute="_compute_opportunities_totals",
+        store=True,
+        help="Open opportunities for events of this category.",
+    )
+    seats_wanted_sum = fields.Integer(
+        string="Wanted seats",
+        compute="_compute_opportunities_totals",
+        store=True,
+        help="Sum of wanted seats in opportunities for events of this category.",
+    )
+    seats_wanted_total = fields.Char(
+        string="Opportunities (seats)",
+        compute="_compute_opportunities_totals",
+        store=True,
+        help="Open opportunities for events of this category (and wanted seats).",
+    )
+
+    def _events_domain(self):
+        """Basic domain to get related events."""
+        return [
+            ("event_type_id", "in", self.ids),
+            # The following domain is the same as upstream's "Upcoming/Running"
+            # filter, which is the default when opening events view. It'd be
+            # more correct to filter for `date_end >= fields.Datetime.now()`,
+            # to exclude events that finished earlier today. However, that
+            # would make the smart button display a different count than the
+            # events when clicking on it, so it seems more user-friendly to
+            # include these events, even if they finished earlier today.
+            ("date_end", ">=", fields.Date.today()),
+        ]
+
+    def _compute_event_totals(self):
+        """Get how many open events and available seats exist."""
+        domain = self._events_domain()
+        types_with_unlimited_seats = (
+            self.env["event.event"]
+            .search(
+                domain + [("seats_limited", "=", False)],
+            )
+            .mapped("event_type_id")
+        )
+        results = self.env["event.event"].read_group(
+            domain=domain,
+            fields=["seats_available"],
+            groupby=["event_type_id"],
+        )
+        for group in results:
+            event_type = group["event_type_id"][0]
+            events = self.env["event.event"].search(
+                [
+                    ("event_type_id", "=", event_type),
+                    ("date_end", ">=", fields.Date.today()),
+                ]
+            )
+            total_seats = sum(event.seats_available for event in events)
+            group["seats_available"] = total_seats
```

Comment:
IMO this should not be necessary, the `seats_available` field is already being used in read_group and the domain as well.

Reply:
So the solution is to remove the `read_group` for the totals calculation because the `seats_available` field in v17 is not stored and operates directly on data stored in the database.